src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
+            Node<T> currentNode = getNodeByIndex(index);
+            newNode.next = currentNode;
+            newNode.prev = currentNode.prev;
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T elements : list) {
+            add(elements);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getNodeByIndex(index);
+        T result = currentNode.value;
+        currentNode.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        if (index == 0) {
+            T removedElement = head.value;
+            head = head.next;
+            size--;
+            return removedElement;
+        } else if (index == size - 1) {
+            T removedElement = (T) tail.value;
+            tail = tail.prev;
+            size--;
+            return removedElement;
+        } else {
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndexByNode(object);
+        if (index == -1) {
+            return false;
+        } else {
+            remove(index);
+            return true;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Index out of bounds");
+        }
+    }
+
+    private void checkAddedIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> current;
+        if (index < size / 2) {
+            current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+        } else {
+            current = tail;
+            for (int i = size - 1; i > index; i--) {
+                current = current.prev;
+            }
+        }
+        return current;
+    }
+
+    private int getIndexByNode(T value) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if ((value == null && null == node.value)
+                    || (value != null && value.equals(node.value))) {
```

Comment:
```suggestion
            if (value == null && null == node.value
                    || value != null && value.equals(node.value)) {
```
Redundant  brackets  (read about operators' priority)